src/Nethermind/Nethermind.Verkle.Tree/VerkleTree.Prover.cs

```diff
@@ -63,6 +63,32 @@ public ExecutionWitness GenerateExecutionWitness(byte[][] keys, out Banderwagon
         return new ExecutionWitness(stemStateDiffList, proof);
     }
 
+    public void UpdateExecutionWitness(ExecutionWitness executionWitness, byte[][] keys)
+    {
+        SpanDictionary<byte, List<SuffixStateDiff>> stemStateDiffWithNewValue = new(Bytes.SpanEqualityComparer);
+        foreach (var key in keys)
+        {
+            SuffixStateDiff suffixData = new() { NewValue = Get(key) };
+            Span<byte> keyStem = key.AsSpan()[..31];
+            if (!stemStateDiffWithNewValue.TryGetValue(keyStem, out List<SuffixStateDiff> suffixStateDiffList))
+            {
+                suffixStateDiffList = [];
+                stemStateDiffWithNewValue.TryAdd(keyStem, suffixStateDiffList);
+            }
+            suffixStateDiffList.Add(suffixData);
+        }
+        executionWitness.StateDiff.ForEach(stateDiff =>
+        {
+            List<SuffixStateDiff> newValueSuffixDiffs = stemStateDiffWithNewValue[stateDiff.Stem.Bytes];
+            int i = 0;
+            stateDiff.SuffixDiffs.ForEach(suffixDiffs =>
+            {
+                suffixDiffs.NewValue = newValueSuffixDiffs[i].NewValue;
+                i++;
+            });
+        });
+    }
```

Comment:
you can just directly iterate over executionWitness and then query value and assign it directly there. 
no need to allocate a new dictionary.